js/profile.js

```diff
@@ -0,0 +1,211 @@
+
+    userPostsContainer.innerHTML = '';
+
+    // Loop through the posts and create HTML elements for each post
+    posts.forEach((post) => {
+        const postElement = document.createElement('div');
+        postElement.classList.add('row', 'mb-3');
+
+        // Replace backslashes with HTML line break tags
+        const postTitleWithLineBreaks = post.title.replace(/\n/g, '<br>');
+
+        postElement.innerHTML = `
+        <div class="col-md-12 bg-white p-4 rounded shadow mb-4 mt-4">
+          
+          <p class="text-secondary">${formatDate(post.created)}</p>
+          <p class="fs-4">${authorName}</p>
+          <div class="shadow p-2">
+          <p class="text-primary-emphasis">${postTitleWithLineBreaks}</p>
+          <p class="post-content">${post.body}</p>
+          ${post.media ? `<img src="${post.media}" alt="Post Image" class="img-fluid">` : ''}
+          </div>
+          <button class="btn btn-primary" onclick="editPost(${post.id})">Edit</button>
+          <button class="btn btn-danger" onclick="deletePost(${post.id})">Delete</button>
+
+        </div>
+      `;
+      
+        userPostsContainer.appendChild(postElement);
+    });
+}
+
+// Call the function to fetch and display posts of the specific author
+fetchAuthorPosts(authorName);
+
+
+// Function to edit a post
+function editPost(postId) {
+    const isProfilePage = window.location.pathname === '/profile/index.html';
+
+    if (isProfilePage) {
+        // Check if there are no query parameters in the URL
+        const hasNoQueryParams = window.location.search === '';
+
+        if (hasNoQueryParams) {
+            // Find the post with the given postId
+            const postToEdit = postsData.find(post => post.id === postId);
+
+            if (!postToEdit) {
+                alert('Post not found for editing');
+                return;
+            }
+
+            // Show a prompt for the user to edit the post content
+            const updatedContent = prompt('Edit the post content:', postToEdit.body);
+
+            if (updatedContent === null) {
+                return;
+            }
+
+            // Send a PUT request to update the post
+            updatePost(postId, updatedContent);
+        } else {
+            alert('You can only edit your own posts.');
+        }
+    } else {
+        alert('You can only edit your own posts.');
+    }
+}
+  
+  // Function to update a post via a PUT request
+  async function updatePost(postId, updatedContent) {
+    try {
+      const token = localStorage.getItem('accessToken');
+      const putData = {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+        body: JSON.stringify({ body: updatedContent }),
+      };
+  
+      const response = await fetch(`${API_BASE_URL}/api/v1/social/posts/${postId}`, putData);
+  
+      if (response.ok) {
+        console.log('Post updated successfully');
```

Comment:
You can show an alert to inform the user that the operation has been successfully completed